specification/under_development/profiles/Base_Profile.md

```diff
@@ -0,0 +1,522 @@
+  program logic
+- declarations of the [QIS functions](#quantum-instruction-set) used by the
+  program
+- declarations of [runtime functions](#runtime-functions) used for
+  initialization and output recording
+- one or more [attribute groups](#attributes) used to store information about
+  the entry point, and optionally additional information about other function
+  declarations
+- [module flags](#module-flags-metadata) that contain information that a
+  compiler or backend may need to process the bitcode.
+
+The human readable LLVM IR for the bitcode can be obtained using standard [LLVM
+tools](https://llvm.org/docs/CommandGuide/llvm-dis.html). For the purpose of
+clarity, this specification contains examples of the human readable IR emitted
+by [LLVM 13](https://releases.llvm.org/13.0.1/docs/LangRef.html). While the
+bitcode representation is portable and usually backward compatible, there may be
+visual differences in the human readable format depending on the LLVM version.
+These differences are irrelevant when using standard tools to load, manipulate,
+and/or execute bitcode.
+
+The code below illustrates how a simple program looks within a Base Profile
+representation:
+
+```llvm
+; type definitions
+
+%Result = type opaque
+%Qubit = type opaque
+
+; global constants (labels for output recording)
+
+@0 = internal constant [3 x i8] c"r1\00"
+@1 = internal constant [3 x i8] c"r2\00"
+
+; entry point definition
+
+define i64 @Entry_Point_Name() #0 {
+entry:
+  ; calls to initialize the execution environment
+  call void @__quantum__rt__initialize(i8* null)
+  br label %body
+
+body:                                     ; preds = %entry
+  ; calls to QIS functions that are not irreversible
+  call void @__quantum__qis__h__body(%Qubit* null)
+  call void @__quantum__qis__cnot__body(%Qubit* null, %Qubit* inttoptr (i64 1 to %Qubit*))
+  br label %measurements
+
+measurements:                             ; preds = %body
+  ; calls to QIS functions that are irreversible
+  call void @__quantum__qis__mz__body(%Qubit* null, writeonly %Result* null)
+  call void @__quantum__qis__mz__body(%Qubit* inttoptr (i64 1 to %Qubit*), writeonly %Result* inttoptr (i64 1 to %Result*))
+  br label %output
+
+output:                                   ; preds = %measurements
+  ; calls to record the program output
+  call void @__quantum__rt__tuple_record_output(i64 2, i8* null)
+  call void @__quantum__rt__result_record_output(%Result* null, i8* getelementptr inbounds ([3 x i8], [3 x i8]* @0, i32 0, i32 0))
+  call void @__quantum__rt__result_record_output(%Result* inttoptr (i64 1 to %Result*), i8* getelementptr inbounds ([3 x i8], [3 x i8]* @1, i32 0, i32 0))
+
+  ret i64 0
+}
+
+; declarations of QIS functions
+
+declare void @__quantum__qis__h__body(%Qubit*)
+
+declare void @__quantum__qis__cnot__body(%Qubit*, %Qubit*)
+
+declare void @__quantum__qis__mz__body(%Qubit*, writeonly %Result*) #1
```

Comment:
```suggestion
measurements:                             ; preds = %body
  ; calls to QIS functions that are irreversible
  call void @__quantum__qis__mz__body(%Qubit* null, %Result* writeonly null)
  call void @__quantum__qis__mz__body(%Qubit* inttoptr (i64 1 to %Qubit*), %Result* writeonly inttoptr (i64 1 to %Result*))
  br label %output

output:                                   ; preds = %measurements
  ; calls to record the program output
  call void @__quantum__rt__tuple_record_output(i64 2, i8* null)
  call void @__quantum__rt__result_record_output(%Result* null, i8* getelementptr inbounds ([3 x i8], [3 x i8]* @0, i32 0, i32 0))
  call void @__quantum__rt__result_record_output(%Result* inttoptr (i64 1 to %Result*), i8* getelementptr inbounds ([3 x i8], [3 x i8]* @1, i32 0, i32 0))

  ret i64 0
}

; declarations of QIS functions

declare void @__quantum__qis__h__body(%Qubit*)

declare void @__quantum__qis__cnot__body(%Qubit*, %Qubit*)

declare void @__quantum__qis__mz__body(%Qubit*, %Result* writeonly) #1
```

This was required for successful assembly with llvm 14.0.6; llvm 13.0.0 and 13.0.1 also did not like `writeonly` where it was.

Reply:
Thanks! I meant to confirm that and didn't get back to it...